frontend/app/components/address.tsx

```diff
@@ -0,0 +1,30 @@
+import type { ComponentProps } from 'react';
+
+import clsx from 'clsx';
+
+export interface AddressProps extends ComponentProps<'address'> {
+  address: string;
+  city: string;
+  provinceState?: string;
+  postalZipCode?: string;
+  country: string;
+}
+
+export function Address(props: AddressProps) {
+  const { address, city, provinceState, postalZipCode, country, className, ...restProps } = props;
+
+  // TODO 'canada' shouldn't be hardcoded as we may deal with different values of the country field such as abbreviations
+  const isNotCanadianAddress = 'canada' !== country.toLowerCase();
+
+  return (
+    <address className={clsx('mb-0', className)} {...restProps}>
+      <p className="mb-0">{address}</p>
+      <p className="mb-0">
+        {city}
+        {provinceState && ` ${provinceState}`}
+        {postalZipCode && <span className="whitespace-pre-wrap">{`  ${postalZipCode}`}</span>}
+      </p>
+      {isNotCanadianAddress && <p className="mb-0">{country}</p>}
```

Comment:
Create a utility function that will format the address as the Canada Post addressing guideline for International addresses.

```ts
function formatAddress({ unitNum, street, city, state, zipCode, country }: Address): string {
  const lines = [`${unitNum ?? ''} ${street ?? ''}`,
      `${city ?? ''} ${state ?? ''}  ${zipCode ?? ''}`,
      `${country ?? ''}`];
  return lines.map((line) => line.trim()).filter(Boolean).join('\n');
}
```

Take a look [here](https://www.typescriptlang.org/play?ssl=15&ssc=2&pln=10&pc=1#code/JYOwLgpgTgZghgYwgAgIIBN1QgZx8gbwChlkBXEYMAOTIFsB+ALmRzClAHMBuE19iBDDN+HEDz4IqATxFsxE0mziQ57Lr1IAvYAAcAwgHt0ENQs3IEhiu1kt5GogF8iRGBQRhghkMhiGoOhUMLFwcAAoCckoaegAaUUEwBKkwaQTlSASdA2MIFOtwKGlkJxYQ7DwASnt1cUJJHzZkABtQXGQAXmQAbQADABICCipaOmQGBmQAcmmnZCH5JImp2ac+uL5SUkGCVJLJmbmFgkyUQ7XSIZyjExWj9c3t7d2rG2L7tb6AXQtsMDIUF8bRAuAAdEFdOFwiCIFUugA+VrtMHqOjhKpVMEwYAtSBQcIAIUMhhaEDgICxACtDKBwtMADogaZVXguIgAeg5yAAogAPOB0XRk5AAVRwcE4ECYRCsIGaEqFZIqYXKmEq+G6xFII1idBY0wAjAAmADM0yeiSEBoAsnBQMgAMpgC2SGQG-QyABuuLJrqUYBU0pm+lQ-uQNzyBpNpoALABWcNvIrSD2FWwtCnoabOXiypqkiBglqGTjhfyBYLqsLhRXCiAq6qsoA).

Reply:
Fixed. 

I had to modify a bit of the suggested `formatAddress(..)` to remove any unnecessary spaces as a result of using `whitespace-pre-wrap`.